src/components/document/annotation-layer.tsx

```diff
@@ -0,0 +1,173 @@
+import classNames from "classnames";
+import { observer } from "mobx-react";
+import React, { useContext, useEffect, useState } from "react";
+
+import { ArrowAnnotationComponent } from "../annotations/arrow-annotation";
+import { TileApiInterfaceContext } from "../tiles/tile-api";
+import { useUIStore } from "../../hooks/use-stores";
+import { ArrowAnnotation, ArrowAnnotationType } from "../../models/annotations/arrow-annotation";
+import { ObjectBoundingBox, ClueObjectModel, ClueObjectType } from "../../models/annotations/clue-object";
+import { DocumentContentModelType } from "../../models/document/document-content";
+
+import "./annotation-layer.scss";
+
+interface IAnnotationButtonProps {
+  getObjectBoundingBox: (rowId: string, tileId: string, objectId: string) => ObjectBoundingBox | undefined;
+  key?: string;
+  objectId: string;
+  objectType?: string;
+  onClick?: (tileId: string, objectId: string, objectType?: string) => void;
+  rowId: string;
+  sourceObjectId?: string;
+  sourceTileId?: string;
+  tileId: string;
+}
+const AnnotationButton = observer(function AnnotationButton({
+  getObjectBoundingBox, objectId, objectType, onClick, rowId, sourceObjectId, sourceTileId, tileId
+}: IAnnotationButtonProps) {
+  const style = getObjectBoundingBox(rowId, tileId, objectId);
+  if (!style) return null;
+
+  const handleClick = () => onClick?.(tileId, objectId, objectType);
+
+  const source = sourceObjectId === objectId && sourceTileId === tileId;
+  const classes = classNames("annotation-button", { source });
+  return (
+    <button className={classes} onClick={handleClick} style={style} />
+  );
+});
+
+interface IAnnotationLayerProps {
+  content?: DocumentContentModelType;
+  documentScrollX?: number;
+  documentScrollY?: number;
+  readOnly?: boolean;
+}
+export const AnnotationLayer = observer(function AnnotationLayer({
+  content, documentScrollX, documentScrollY, readOnly
+}: IAnnotationLayerProps) {
+  const [_initialized, setInitialized] = useState(false);
+  useEffect(() => {
+    // Forces the annotation layer to rerender after initial load, getting access to the locations of elements.
+    setTimeout(() => setInitialized(true));
```

Comment:
Out of curiosity, is the `setTimeout()` necessary? The effect should only run after the initial render, so I would expect the result to be the same without it.

Reply:
It was not necessary, good catch 👍